libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewStyles.kt

```diff
@@ -49,6 +49,12 @@ internal class NavigationViewStyles(context: Context) {
         MutableStateFlow(ViewStyleCustomization.defaultMarkerAnnotationOptions(context))
     private var _arrivalTextAppearance: MutableStateFlow<Int> =
         MutableStateFlow(ViewStyleCustomization.defaultArrivalTextAppearance())
+    private var _mapCompassParams: MutableStateFlow<MapboxMapCompassParams> = MutableStateFlow(
```

Comment:
Why not directly use `CompassSettingsInterface` instead.

```
private var _mapCompass: MutableStateFlow<CompassSettingsInterface> = MutableStateFlow(defaultCompass())
```

Same applies to `scalebar`.

This way we can avoid creating additional builder which is just a wrapper around this interface and eventually will be supporting all the properties the interface does.

@dzinad

Reply:
But do we want to expose all the parameters now? I thought we wanted only the enable flag. Otherwise it would get messy, especially with our current limitations to where it can be placed.

Reply:
That's true, but if we go with the builder we would have to maintain it until next stable release. How about we use `CameraSettingsInterface`, but not apply it in `CompassComponent`. In the `CompassComponent`, we can do as follows:

```
class CompassComponent(
    ....,
    val compassSettings: StateFlow<CompassSettingsInterface>
) {
    init {
        mapView.compass.updateSettings {
            enabled = compassSettings.enabled
            // we won't apply other settings
        }
    }
}
```
We can document that the only settings applied will be `enabled` as you have already done. This way we can support other parameters whenever we want
Wdyt?

@dzinad

Reply:
But we can add the parameters to builder whenever we want too and that won't break API.
I'm not really in favour of using `CameraSettingsInterface` because:
1) We probably won't support all the parameters anyway;
2) Not including the parameter in the API is much more clear than documenting that it does not work.